plugins/inputs/pgbouncer/pgbouncer.go

```diff
@@ -20,23 +20,108 @@ var sampleConfig string
 
 type PgBouncer struct {
 	postgresql.Service
+	ShowCommands []string `toml:"show_commands"`
 }
 
 var ignoredColumns = map[string]bool{"user": true, "database": true, "pool_mode": true,
 	"avg_req": true, "avg_recv": true, "avg_sent": true, "avg_query": true,
+	"force_user": true, "host": true, "port": true, "name": true,
 }
 
 func (*PgBouncer) SampleConfig() string {
 	return sampleConfig
 }
 
 func (p *PgBouncer) Gather(acc telegraf.Accumulator) error {
+	if len(p.ShowCommands) == 0 {
+		if err := p.showStats(acc); err != nil {
+			return err
+		}
+
+		if err := p.showPools(acc); err != nil {
+			return err
+		}
+	} else {
+		for _, cmd := range p.ShowCommands {
+			switch {
+			case cmd == "stats":
+				if err := p.showStats(acc); err != nil {
+					return err
+				}
+			case cmd == "pools":
+				if err := p.showPools(acc); err != nil {
+					return err
+				}
+			case cmd == "lists":
+				if err := p.showLists(acc); err != nil {
+					return err
+				}
+			case cmd == "databases":
+				if err := p.showDatabase(acc); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+type scanner interface {
+	Scan(dest ...interface{}) error
+}
+
+func (p *PgBouncer) accRow(row scanner, columns []string) (map[string]string,
+	map[string]*interface{}, error) {
```

Comment:
Was this line too long and that's why you broke it?

Reply:
Thanks for the comments. I tried to do it so as not to change the existing code, so I made a minimum of changes. This construction was in the old code, I decided not to touch it. I'll try to correct her.